### ISSUE TITLE
Unhandled 500 on Login

### ISSUE THREAD

**Issue by ZigZagT** [bitwarden/server#3008]
### Steps To Reproduce

1. Deploy the unified container (see the docker-compose fragment below)
2. Create an account
3. Login with the account just created
4. Shows an error and login is unsuccess.

### Expected Result

Login successfully

### Actual Result

Login not successful

### Additional Context

The error is due to a 500 returned from `/identity/connect/token`

Error logs from `var/log/bitwarden/identity.log`:

```
crit: IdentityServer4.Hosting.IdentityServerMiddleware[0]
      => SpanId:050b64273b895c32, TraceId:643d957a602be558268094873787e001, ParentId:0000000000000000 => ConnectionId:0HMR8Q34PECPD => RequestPath:/identity/connect/token RequestId:0HMR8Q34PECPD:00000002
      Unhandled exception: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
      System.FormatException: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
         at System.Guid.GuidResult.SetFailure(Boolean overflow, String failureMessageID)
         at System.Guid.TryParseExactN(ReadOnlySpan`1 guidString, GuidResult& result)
         at System.Guid.TryParseGuid(ReadOnlySpan`1 guidString, GuidResult& result)
         at System.Guid..ctor(String g)
         at Bit.Identity.IdentityServer.ClientStore.CreateApiKeyClientAsync(String clientId) in /source/src/Identity/IdentityServer/ClientStore.cs:line 92
         at Bit.Identity.IdentityServer.ClientStore.FindClientByIdAsync(String clientId) in /source/src/Identity/IdentityServer/ClientStore.cs:line 87
         at IdentityServer4.Stores.ValidatingClientStore`1.FindClientByIdAsync(String clientId)
         at IdentityServer4.Stores.IClientStoreExtensions.FindEnabledClientByIdAsync(IClientStore store, String clientId)
         at IdentityServer4.Validation.ClientSecretValidator.ValidateAsync(HttpContext context)
         at IdentityServer4.Endpoints.TokenEndpoint.ProcessTokenRequestAsync(HttpContext context)
         at IdentityServer4.Endpoints.TokenEndpoint.ProcessAsync(HttpContext context)
         at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events, IBackChannelLogoutService backChannelLogoutService)
fail: Microsoft.AspNetCore.Server.Kestrel[13]
      => SpanId:050b64273b895c32, TraceId:643d957a602be558268094873787e001, ParentId:0000000000000000 => ConnectionId:0HMR8Q34PECPD => RequestPath:/identity/connect/token RequestId:0HMR8Q34PECPD:00000002
      Connection id "0HMR8Q34PECPD", Request id "0HMR8Q34PECPD:00000002": An unhandled exception was thrown by the application.
      System.FormatException: Guid should contain 32 digits with 4 dashes (xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx).
         at System.Guid.GuidResult.SetFailure(Boolean overflow, String failureMessageID)
         at System.Guid.TryParseExactN(ReadOnlySpan`1 guidString, GuidResult& result)
         at System.Guid.TryParseGuid(ReadOnlySpan`1 guidString, GuidResult& result)
         at System.Guid..ctor(String g)
         at Bit.Identity.IdentityServer.ClientStore.CreateApiKeyClientAsync(String clientId) in /source/src/Identity/IdentityServer/ClientStore.cs:line 92
         at Bit.Identity.IdentityServer.ClientStore.FindClientByIdAsync(String clientId) in /source/src/Identity/IdentityServer/ClientStore.cs:line 87
         at IdentityServer4.Stores.ValidatingClientStore`1.FindClientByIdAsync(String clientId)
         at IdentityServer4.Stores.IClientStoreExtensions.FindEnabledClientByIdAsync(IClientStore store, String clientId)
         at IdentityServer4.Validation.ClientSecretValidator.ValidateAsync(HttpContext context)
         at IdentityServer4.Endpoints.TokenEndpoint.ProcessTokenRequestAsync(HttpContext context)
         at IdentityServer4.Endpoints.TokenEndpoint.ProcessAsync(HttpContext context)
         at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events, IBackChannelLogoutService backChannelLogoutService)
         at IdentityServer4.Hosting.IdentityServerMiddleware.Invoke(HttpContext context, IEndpointRouter router, IUserSession session, IEventService events, IBackChannelLogoutService backChannelLogoutService)
         at IdentityServer4.Hosting.MutualTlsEndpointMiddleware.Invoke(HttpContext context, IAuthenticationSchemeProvider schemes)
         at Microsoft.AspNetCore.Authentication.AuthenticationMiddleware.Invoke(HttpContext context)
         at IdentityServer4.Hosting.BaseUrlMiddleware.Invoke(HttpContext context)
         at Bit.Core.Utilities.CurrentContextMiddleware.Invoke(HttpContext httpContext, ICurrentContext currentContext, GlobalSettings globalSettings) in /source/src/Core/Utilities/CurrentContextMiddleware.cs:line 19
         at Microsoft.AspNetCore.Localization.RequestLocalizationMiddleware.Invoke(HttpContext context)
         at Bit.SharedWeb.Utilities.ServiceCollectionExtensions.<>c__DisplayClass11_0.<<UseDefaultMiddleware>b__1>d.MoveNext() in /source/src/SharedWeb/Utilities/ServiceCollectionExtensions.cs:line 527
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Builder.Extensions.UsePathBaseMiddleware.InvokeCore(HttpContext context, PathString matchedPath, PathString remainingPath)
         at Bit.Identity.Startup.<>c__DisplayClass10_1.<<Configure>b__2>d.MoveNext() in /source/src/Identity/Startup.cs:line 174
      --- End of stack trace from previous location ---
         at Microsoft.AspNetCore.Server.Kestrel.Core.Internal.Http.HttpProtocol.ProcessRequests[TContext](IHttpApplication`1 application)
```

### Docker compose I was using:

```
  bitwarden:
    image: bitwarden/self-host:beta
    environment:
      - BW_DOMAIN=<redacted>
      - BW_DB_PROVIDER=sqlite
      - BW_DB_FILE=/etc/bitwarden/vault.db
      - BW_INSTALLATION_ID=<redacted>
      - BW_INSTALLATION_KEY=<redacted>
      - PUID=1035
      - PGID=100
      - BW_ENABLE_SSL=false
      - globalSettings__mail__replyToEmail=<redacted>
      - globalSettings__mail__smtp__host=<redacted>
      - globalSettings__mail__smtp__port=465
      - globalSettings__mail__smtp__ssl=true
      - globalSettings__mail__smtp__username=<redacted>
      - globalSettings__mail__smtp__password=<redacted>
    volumes:
      - ./bitwarden:/etc/bitwarden
    restart: always
```

### Githash Version

1202a670-dirty

### Environment Details

Docker ImageID = 86bfac035094

### Issue-Link

https://github.com/bitwarden/server/issues/2480

### Issue Tracking Info

- [ ] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by justindbaur** [comment]
What client are you using to sign into the account (website, browser extension, mobile phone)?

**Comment by ZigZagT** [comment]
> What client are you using to sign into the account (website, browser extension, mobile phone)?

I was using a browser

**Comment by justindbaur** [comment]
@ZigZagT Would you be able to open your dev tools, go to the `Network` tab, then attempt to login and find the `/identity/connect/token` call? I would like to see what the `clientId` in the request looks like.

**Comment by ZigZagT** [comment]
`client_id=web`

Here's the full request & response for reference:

Request:
```
:authority: <redacted>
:method: POST
:path: /identity/connect/token
:scheme: https
accept: application/json
accept-encoding: gzip, deflate, br
accept-language: en-US,en;q=0.9
auth-email: <redacted>
authorization: Basic <redacted>
bitwarden-client-name: web
bitwarden-client-version: 2023.4.2
cache-control: no-cache
content-length: 232
content-type: application/x-www-form-urlencoded; charset=utf-8
device-type: 9
origin: <redacted>
pragma: no-cache
referer: <redacted>
sec-ch-ua: "Not.A/Brand";v="8", "Chromium";v="114", "Google Chrome";v="114"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "macOS"
sec-fetch-dest: empty
sec-fetch-mode: cors
sec-fetch-site: same-origin
user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36

scope=api%2520offline_access&client_id=web&deviceType=9&deviceIdentifier=<redacted>&deviceName=chrome&grant_type=password&username=<redacted>&password=<redacted>
```

Response:
```
alt-svc: h3=":443"; ma=86400
cf-cache-status: DYNAMIC
cf-ray: <redacted>
content-length: 0
date: Mon, 12 Jun 2023 20:44:45 GMT
expect-ct: max-age=86400, enforce
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
referrer-policy: same-origin
report-to: {"endpoints":[{"url":"<redacted>"}],"group":"cf-nel","max_age":604800}
server: cloudflare
strict-transport-security: max-age=15552000; includeSubDomains; preload
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
x-xss-protection: 1; mode=block
```

A couple of things that on top of my mind that might have contributed to the error:
1. The connection is proxied by CloudFlare
2. The server is protected by HTTP basic auth

Still unsure what exactly the cause is, given the error message `System.FormatException: Guid should contain 32 digits with 4 dashes` is way too generic

**Comment by ZigZagT** [comment]
@justindbaur 

Did a few AB-testing, turns out the culprit was HTTP Basic Auth in our setup. The API crashes whenever the `authorization:` header was sent with request.

Question: Should this be classified as a bug and get fixed?

**Comment by ZigZagT** [comment]
@justindbaur 

Thanks for explaining into the details. Really appreciate the transparency. 

Our usage was to block traffic during evaluating the new BW deployment. Now since the root cause is clear, we'll simply drop the HTTP auth config as a workaround.

Also since we were literally using web browsers' HTTP Auth capability, `proxy-auth` just won't cut it.

We'd love to see followup fixes be implemented, as you've mentioned above. I understand making `authorization` header work is complex and might not align with your roadmap. So please at least throw an actionable error message on the UI before calling it done ;)

Anyways, now since we're unblocked, please feel free to close the issue.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
